src/software/simulated_tests/simulated_tactic_test_fixture.cpp

```diff
@@ -5,85 +5,181 @@
 }
 
 void SimulatedTacticTestFixture::updatePrimitives(
+    const World& friendly_world, const World& enemy_world,
+    std::shared_ptr<Simulator> simulator_to_update)
+{
+    updateFriendlyPrimitives(friendly_world, simulator_to_update);
+    if (this->enemy_robot_id)
+    {
+        updateEnemyPrimitives(enemy_world, simulator_to_update);
+    }
+}
+
+void SimulatedTacticTestFixture::updateFriendlyPrimitives(
     const World& world, std::shared_ptr<Simulator> simulator_to_update)
 {
     std::vector<std::unique_ptr<Intent>> intents;
     auto start_tick_time = std::chrono::system_clock::now();
 
-    if (!robot_id)
+    if (!friendly_robot_id)
     {
-        LOG(FATAL) << "No robot id set" << std::endl;
+        LOG(FATAL) << "No friendly robot id set" << std::endl;
     }
-    else if (auto new_robot = world.friendlyTeam().getRobotById(*robot_id))
+    else if (auto new_robot = world.friendlyTeam().getRobotById(*friendly_robot_id))
     {
-        auto intent = tactic->get(*world.friendlyTeam().getRobotById(*robot_id), world);
-        intent->setMotionConstraints(motion_constraints);
+        auto intent = friendly_tactic->get(
+            *world.friendlyTeam().getRobotById(*friendly_robot_id), world);
+        intent->setMotionConstraints(friendly_motion_constraints);
         intents.push_back(std::move(intent));
     }
     else
     {
-        LOG(FATAL) << "No robot with robot id " << *robot_id << std::endl;
+        LOG(FATAL) << "No friendly robot with robot id " << *friendly_robot_id
+                   << std::endl;
     }
 
-    auto primitive_set_msg = navigator->getAssignedPrimitives(world, intents);
+    auto primitive_set_msg = friendly_navigator->getAssignedPrimitives(world, intents);
     double duration_ms     = ::TestUtil::millisecondsSince(start_tick_time);
     registerTickTime(duration_ms);
     simulator_to_update->setYellowRobotPrimitiveSet(
         createNanoPbPrimitiveSet(*primitive_set_msg));
 }
 
+void SimulatedTacticTestFixture::updateEnemyPrimitives(
+    const World& world, std::shared_ptr<Simulator> simulator_to_update)
+{
+    std::vector<std::unique_ptr<Intent>> intents;
+    auto start_tick_time = std::chrono::system_clock::now();
+
+    if (!enemy_robot_id)
+    {
+        LOG(FATAL) << "No enemy robot id set" << std::endl;
+    }
+    else if (auto new_robot = world.friendlyTeam().getRobotById(*enemy_robot_id))
+    {
+        auto intent =
+            enemy_tactic->get(*world.friendlyTeam().getRobotById(*enemy_robot_id), world);
+        intent->setMotionConstraints(enemy_motion_constraints);
+        intents.push_back(std::move(intent));
+    }
+    else
+    {
+        LOG(FATAL) << "No enemy robot with robot id " << *enemy_robot_id << std::endl;
+    }
+
+    auto primitive_set_msg = enemy_navigator->getAssignedPrimitives(world, intents);
+    double duration_ms     = ::TestUtil::millisecondsSince(start_tick_time);
+    registerTickTime(duration_ms);
```

Comment:
perhaps we should have an enemy and friendly tick time?

Reply:
Not sure what advantage we get from that - Wouldn't we want both sets of robots to update on the same tick?

Reply:
if the tick time counts both enemy and friendly, then the information is probably less useful, since you wouldn't know if a long tick time is caused by enemy robots or friendly robots

Reply:
resolved